extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,370 @@
+					CSSStyles: {
+						'font-size': 'large',
+						'line-height': '22pt',
+						'margin-left': '1em',
+						'margin-top': '0em'
+					}
+				}).component();
+
+			this.clearAndAddTransformationContainerComponents(false);
+
+			const flexGrid = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row',
+				height: '100%',
+				width: '100%'
+			}).component();
+
+			/**
+			 * Setting height of the div based on the total viewport height after removing dialog
+			 * header and footer heights. With this the div will occupy the entire page space of the dialog.
+			 */
+			flexGrid.addItem(columnContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'min-height': 'calc(100vh - 160px)'
+				}
+			});
+			flexGrid.addItem(this._transformationContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'overflow': 'scroll',
+					'padding-right': '10px',
+					'width': '900px',
+					'max-height': 'calc(100vh - 160px)'
+				}
+			});
+
+			const formBuilder = view.modelBuilder.formContainer().withFormItems(
+				[
+					{
+						component: flexGrid
+					}
+				],
+				{
+					horizontal: false
+				}
+			);
+			const form = formBuilder.withLayout({ width: '100%' }).component();
+			return view.initializeModel(form);
+		});
+
+		this._dialogObject.okButton.onClick(e => {
+			this._doneEmitter.emit('done');
+		});
+
+		this._dialogObject.cancelButton.onClick(e => {
+			this._doneEmitter.emit('close');
+		});
+
+		this._dialogObject.content = [tab];
+		azdata.window.openDialog(this._dialogObject);
+		return new Promise((resolve) => {
+			this._doneEmitter.once('done', async () => {
+				try {
+					await this._provider.sendSaveTransformationRequest({
+						derivedColumnName: this.currentDerivedColumnName
+					});
+					resolve({
+						derivedColumnName: this.currentDerivedColumnName,
+						derivedColumnDataPreview: this.currentTransformation
+					});
+				} catch (e) {
+					console.log(e); // Need to have better error handling for saved transformation. However this seems to be mostly a non-issue.
+				}
+			});
+
+			this._doneEmitter.once('close', async () => {
+				resolve(undefined);
+				azdata.window.closeDialog(this._dialogObject);
+			});
+		});
+	}
+
+	private clearAndAddTransformationContainerComponents(addTable: boolean): void {
+		this._transformationContainer.clearItems();
+		if (addTable) {
+			this._transformationContainer.addItem(this._specifyDerivedColumnNameContainer, { flex: '0 0 auto' });
+			this._transformationContainer.addItem(this._transformationTable, { flex: '1 1 auto', CSSStyles: { 'overflow': 'scroll' } });
+		}
+		else {
+			this._transformationContainer.addItem(this._headerInstructionText, { flex: '0 0 auto' });
+			this._transformationContainer.addItem(this._bodyInstructionText, { flex: '0 0 auto' });
+		}
+	}
+
+	private validatePage(): void {
+		this._dialogObject.okButton.enabled = this.currentDerivedColumnName !== undefined && this.currentTransformation.length !== 0;
```

Comment:
is the button enabled by default? what if user click on the button directly without making any changes (means this validation won't be triggered)

Reply:
No, this button is not enabled by default. Probably because of the required input column name being empty when the dialog is first open.